src/main/java/com/cloudbees/AWSCredentialsUtils.java

```diff
@@ -0,0 +1,60 @@
+package com.cloudbees.jenkins.plugins.awscredentials;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.regions.DefaultAwsRegionProviderChain;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+
+import hudson.ProxyConfiguration;
+import jenkins.model.Jenkins;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+final class AWSCredentialsUtils {
```

Comment:
I'm not sure this class is really useful. The `buildStsClient` function could be in `AWSCredentialsImpl` class like `getClientConfiguration` was and should have stayed.

```suggestion
/*package*/ final class AWSCredentialsUtils {
```

Reply:
It is useful because `buildStsClient` is used by both `AWSCredentialsImpl` and `AmazonWebServicesCredentialsBinding`.

Reply:
sure, I'm just same that you could have made the function `/*package*/ static` in `AWSCredentialsImpl` as the `getClientConfiguration` was.